lib/iris/tests/unit/analysis/test_MAX_RUN.py

```diff
@@ -0,0 +1,300 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for the :data:`iris.analysis.MAX_RUN` aggregator."""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests  # isort:skip
+
+import numpy as np
+import numpy.ma as ma
+
+from iris._lazy_data import as_concrete_data, as_lazy_data, is_lazy_data
+from iris.analysis import MAX_RUN
+
+
+def bool_func(x):
+    return x == 1
+
+
+class UnmaskedTest(tests.IrisTest):
+    def setUp(self):
+        """
+        Set up 1d and 2d unmasked data arrays for max run testing.
+
+        Uses 1 and 3 rather than 1 and 0 to check that lambda is being applied.
+        """
+
+        self.data_1ds = [
+            (np.array([3, 1, 1, 3, 3, 3]), 2),  # One run
+            (np.array([3, 1, 1, 3, 1, 3]), 2),  # Two runs
+            (np.array([3, 3, 3, 3, 3, 3]), 0),  # No run
+            (np.array([3, 3, 1, 3, 3, 3]), 1),  # Max run of 1
+            (np.array([1, 1, 1, 3, 1, 3]), 3),  # Run to start
+            (np.array([3, 1, 3, 1, 1, 1]), 3),  # Run to end
+            (np.array([1, 1, 1, 1, 1, 1]), 6),  # All run
+        ]
+
+        self.data_2d_axis0 = np.array(
+            [
+                [3, 1, 1, 3, 3, 3],  # One run
+                [3, 1, 1, 3, 1, 3],  # Two runs
+                [3, 3, 3, 3, 3, 3],  # No run
+                [3, 3, 1, 3, 3, 3],  # Max run of 1
+                [1, 1, 1, 3, 1, 3],  # Run to start
+                [3, 1, 3, 1, 1, 1],  # Run to end
+                [1, 1, 1, 1, 1, 1],  # All run
+            ]
+        ).T
+        self.expected_2d_axis0 = np.array([2, 2, 0, 1, 3, 3, 6])
+
+        self.data_2d_axis1 = self.data_2d_axis0.T
+        self.expected_2d_axis1 = self.expected_2d_axis0
+
+
+class MaskedTest(tests.IrisTest):
+    def setUp(self):
+        """
+        Set up 1d and 2d unmasked data arrays for max run testing.
+
+        Uses 1 and 3 rather than 1 and 0 to check that lambda is being applied.
+        """
+
+        self.data_1ds = [
+            (
+                ma.masked_array(
+                    np.array([1, 1, 1, 3, 1, 3]), np.array([0, 0, 0, 0, 0, 0])
+                ),
+                3,
+            ),  # No mask
+            (
+                ma.masked_array(
+                    np.array([1, 1, 1, 3, 1, 3]), np.array([0, 0, 0, 0, 1, 1])
+                ),
+                3,
+            ),  # Mask misses run
+            (
+                ma.masked_array(
+                    np.array([1, 1, 1, 3, 1, 3]), np.array([1, 1, 1, 0, 0, 0])
+                ),
+                1,
+            ),  # Mask max run
+            (
+                ma.masked_array(
+                    np.array([1, 1, 1, 3, 1, 3]), np.array([0, 0, 1, 0, 0, 0])
+                ),
+                2,
+            ),  # Partially mask run
```

Comment:
What happens when a mask interupts the middle of a run?
e.g.
```
ma.masked_array(
                    np.array([1, 1, 1, 3, 1, 3]), np.array([0, 1, 0, 0, 0, 0])
                )
```

Reply:
I would expect a masked point to end a run, which is the current behaviour (though I'll put in a test for that). Do you think that should be the behaviour?

Reply:
I'm not sure masked points do end a run, at least not with the tests I've tried.

Reply:
I concur, will fix